dvc/repo/imp_url.py

```diff
@@ -10,7 +8,7 @@ def imp_url(
 ):
     from dvc.stage import Stage
 
-    default_out = os.path.basename(urlparse(url).path)
+    default_out = URLInfo(url).name
     out = out or default_out
```

Comment:
`out = out or URLInfo(url).name`
No point in parsing it if we have out specified.